models/accm.py

```diff
@@ -47,6 +58,14 @@ def forward(self, x):
             rsacm_output = self.rsacm_list[i](x_copy)
             self.x_list.append(rsacm_output + x.detach()) # Detach x before adding it to the sum 
             x = rsacm_output + x
+        self.x_list_tensor = torch.stack((self.x_list))
+        if self.args.train_gating_networks:
+            lists_of_outputs = []
+            for index2, elements in enumerate(gating_network_output_index):
+                lists_of_outputs.append(self.x_list_tensor.permute(1,0,2,3,4)[index2][elements])

```

Comment:
We should multiply (select) the x_list_tensor by the gating_network_outputs (one-hot) vectors instead of using gating_network_output_index. Can you print out the shapes of both these? They should be helpful in writing this line of code.

Reply:
gating_network_output_index - shape ([batch_size]) (16)
gating_network_outputs  - shape ([batch_size, self.number_of_rsacm]) (16, 8)

Reply:
I mean the shape of features, outputs of RSACM. Let's say that the outputs of RSACMs is of shape (batch_size, number_of_rsacms, *features_size), for example (64, 8, 50, 16, 16). We want to multiply these by gating_networks_output with shape (batch_size, number_of_rsacms). For me einsum seems to be the simples solution, something like:
`result = torch.einsum('nrchw,nr->nchw', rsacm_outputs, gating_network_outputs)`
Not sure if this is correct, I wrote it on the go. Can you test it?

Reply:
It worked. Thanks!